src/models/User.ts

```diff
@@ -1,28 +1,32 @@
+import {
+  Grade, Role, ScholarshipType, Gender,
+} from './enums';
+
 interface UserProps {
-    age: number;
-    schoolYear: number;
-    role: string;
-    gender: string;
-    scholarshipPreferences: {
-        type: string,
-        major: string,
-        amount: number, 
-        school: string,
-        grade: number,
-        location: string
-    };
-    otherSettings: {
-        notifications: boolean,
-        deadlineNotifications: boolean,
-        relatedNotifications: boolean
-    }
+  age: number;
+  schoolYear: Grade;
+  role: Role;
+  gender: Gender;
+  scholarshipPreferences: {
+    type: ScholarshipType,
+    major: string,
+    amount: number,
+    school: string,
+    schoolYear: Grade,
+    location: string
+  };
+  otherSettings: {
+    notifications: boolean,
+    deadlineNotifications: boolean,
+    relatedNotifications: boolean
+  }
 }
 
-export default class User{
-    id?:string;
-    data: UserProps
+export default class User {
+  id?:string;
+  data: UserProps;
 
-    constructor(data:UserProps){
-        this.data = { ...data };
-    }
-}
\ No newline at end of file
+  constructor(data:UserProps) {
+    this.data = { ...data };
+  }
```

Comment:
Hmm, I think this constructor should be private as it doesn't make sense for to create users. Rather, what we can do is have a `getCurrentUser()` method that loads data based on [`firebase.auth().currentUser`](https://firebase.google.com/docs/auth/web/manage-users#get_the_currently_signed-in_user) and another `get(id: string)` to fetch data for a different user.